### PR TITLE
Utils Cleanup

### PR DESCRIPTION
**Pull request by jjatie**
Replaced unnecessary getters with proper access control
Replaced unnecessary convenience inits with default parameters
Minor refactoring

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -3,7 +3,11 @@ def type
 end
 
 def project_name
-  'Charts.xcodeproj'
+  'ChartsDemo/ChartsDemo.xcodeproj'
```
Comment (liuxuan30) [inline Rakefile]: I think we have merged master into 4.0 already? But these jumps out again

**Comment by jjatie** [comment]
FYI, new PRs from me will be rebasing. I'm scared to do it with the current ones since they switched from master -> 4.0.0.

File: Rakefile (hunk 1 above)
Comment (jjatie) [reply Rakefile]: Just haven't pulled from 4.0.0 in a while.

**Comment by liuxuan30** [comment]
what you mean? I thought since 4.0 is  already sync with master, the rest of PRs can easily pull 4.0 again and simply resolve conflicts.

File: Source/Charts/Utils/Transformer.swift [hunk 2]
```diff
@@ -17,50 +17,50 @@ import CoreGraphics
...
         {
             scaleY = 0.0
         }
 
         // setup all matrices
-        _matrixValueToPx = CGAffineTransform.identity
-        _matrixValueToPx = _matrixValueToPx.scaledBy(x: scaleX, y: -scaleY)
-        _matrixValueToPx = _matrixValueToPx.translatedBy(x: CGFloat(-chartXMin), y: CGFloat(-chartYMin))
+        matrixValueToPx = CGAffineTransform.identity
+            .scaledBy(x: scaleX, y: -scaleY)
```
Comment (liuxuan30) [inline Source/Charts/Utils/Transformer.swift]: is this a chained call? how about putting in on line. seems a little distractive

File: Source/Charts/Utils/ViewPortHandler.swift [hunk 3]
```diff
@@ -250,160 +227,91 @@ open class ViewPortHandler: NSObject
...
-        
-        if newValue == 0.0
-        {
-            newValue = CGFloat.greatestFiniteMagnitude
-        }
-        
-        _maxScaleX = newValue
-        
-        limitTransAndScale(matrix: &_touchMatrix, content: _contentRect)
+        maxScaleX = xScale == 0 ? .greatestFiniteMagnitude : xScale
```
Comment (liuxuan30) [inline Source/Charts/Utils/ViewPortHandler.swift]: is it a  good practice to use 0 as hint to  CGFloat?

File: Source/Charts/Utils/ViewPortHandler.swift [hunk 4]
```diff
@@ -250,160 +227,91 @@ open class ViewPortHandler: NSObject
...
         
         chart.setNeedsDisplay()
         
-        return _touchMatrix
+        return touchMatrix
     }
     
     /// limits the maximum scale and X translation of the given matrix
-    private func limitTransAndScale(matrix: inout CGAffineTransform, content: CGRect?)
+    private func limitTransAndScale(matrix: inout CGAffineTransform, content: CGRect)
```
Comment (liuxuan30) [inline Source/Charts/Utils/ViewPortHandler.swift]: I'm not on my mac now, are we sure all `limitTransAndScale` callers only pass `self.contentRect`?

File: Source/Charts/Utils/Transformer.swift (hunk 2 above)
Comment (jjatie) [reply Source/Charts/Utils/Transformer.swift]: This is a fairly standard Swift practice.

File: Source/Charts/Utils/ViewPortHandler.swift (hunk 4 above)
Comment (jjatie) [reply Source/Charts/Utils/ViewPortHandler.swift]: I am sure.

File: Source/Charts/Utils/ViewPortHandler.swift (hunk 3 above)
Comment (jjatie) [reply Source/Charts/Utils/ViewPortHandler.swift]: `0 == 0.0`. I think the reason `0.0` is used elsewhere is carryover from Objective-C

Comment (liuxuan30) [reply Source/Charts/Utils/ViewPortHandler.swift]: yes, 0.0 looks like more natural as float, while 0 gives me an int imply..

**Pull request merged**